runtime/jcl/common/com_ibm_oti_vm_VM.c

```diff
@@ -156,3 +156,15 @@ Java_com_ibm_oti_vm_VM_markCurrentThreadAsSystemImpl(JNIEnv *env)
 
 	return rc;
 }
+
+/**
+ * Gets the J9ConstantPool address from a J9Class address
+ * @param j9clazz J9Class address
+ * @return address of J9ConstantPool
+ */
+jlong JNICALL
+Java_com_ibm_oti_vm_VM_getJ9ConstantPoolFromJ9Class(JNIEnv *env, jclass unused, jlong j9clazz)
+{
+	J9Class *clazz = (J9Class *)(IDATA)j9clazz;
+	return (jlong)(IDATA)clazz->ramConstantPool;
```

Comment:
also here you need to use a c-style re-interprete caste to avoid the sign extension

some like 

```
*(jlong*)&clazz->ramConstantPool; //you might need to cast to (void*) first, make sure you try this in windows as well
```